File: pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");

    int letters = 0;
    int words = 1; // no space after the last word
    int sentences = 0;
    int len = strlen(text);

    for (int i = 0; i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (isblank(text[i]))
        {
            // there's a space after each word, except the last word
            words++;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }

    // l: the average number of letters per 100 words in the text
    // s: the average number of sentences per 100 words in the text
    float l = (float) letters / words * 100;
    float s = (float) sentences / words * 100;

    // formula for computing the reading level of a text:
    int index = (int) round(0.0588 * l - 0.296 * s - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: pset6/sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
text += " "

letter_count = 0
word_count = 0
sentence_count = 0

for i in range(0, len(text)):
    if text[i].isalpha():
        letter_count += 1
    elif text[i] in [".", "?", "!"]:
        sentence_count += 1
    elif text[i] == " ":
        word_count += 1

# Use Coleman Liau index to compute readability
L = letter_count / word_count * 100
S = sentence_count / word_count * 100
index = int(round(0.0588 * L - 0.296 * S - 15.8))

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
